modules/apps/dispatch/dispatch-talend-web/src/main/java/com/liferay/dispatch/talend/web/internal/executor/TalendDispatchTaskExecutor.java

```diff
@@ -119,6 +169,46 @@ protected TalendArchive fetchTalendArchive(long dispatchTriggerId)
 		return _talendArchiveParser.parse(fileEntry.getContentStream());
 	}
 
+	private String _createClasspath(
+		Path dirPath, DirectoryStream.Filter<Path> filter) {
+
+		try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(
+				dirPath, filter)) {
+
+			StringBundler sb = new StringBundler();
+
+			directoryStream.forEach(
+				path -> {
+					sb.append(path);
+					sb.append(File.pathSeparator);
+				});
+
+			if (sb.index() > 0) {
+				sb.setIndex(sb.index() - 1);
+			}
+
+			return sb.toString();
+		}
+		catch (IOException ioException) {
+			throw new RuntimeException(
+				"Unable to iterate " + dirPath, ioException);
+		}
+	}
+
+	private String _getBootstrapClassPath() {
```

Comment:
Hi @shuyangzhou you instructed me to check/use com.liferay.portal.util.PortalClassPathUtil but it is in portal-impl and it is not available outside.
Do you know if there is another util that can help me to reduce code in this class?? I copy pasted code from Sidecar and I would rather like one specialized Util that can benefit to Sidecar, Dispatch and other future modules.

Reply:
You can use PortalClassPathUtil just fine, portal-impl utils are available from modules.